src/main/java/org/quantumbadger/redreader/activities/OptionsMenuUtility.java

```diff
@@ -377,104 +445,152 @@ public boolean onMenuItemClick(final MenuItem item) {
 							}
 						});
 
-				refreshComments.setShowAsAction(MenuItem.SHOW_AS_ACTION_ALWAYS);
+				refreshComments.setShowAsAction(showAsAction);
 				if(!longText) refreshComments.setIcon(R.drawable.ic_refresh_dark);
 
 				break;
 
 			case PAST_POSTS:
-				menu.add(activity.getString(longText ? R.string.options_past_posts : R.string.options_past))
+				final MenuItem pastPosts = menu.add(activity.getString(longText ? R.string.options_past_posts : R.string.options_past))
 						.setOnMenuItemClickListener(new MenuItem.OnMenuItemClickListener() {
 							public boolean onMenuItemClick(final MenuItem item) {
 								((OptionsMenuPostsListener)activity).onPastPosts();
 								return true;
 							}
 						});
+
+				if(showAsAction != MenuItem.SHOW_AS_ACTION_NEVER) {
+					pastPosts.setShowAsAction(showAsAction);
+					pastPosts.setIcon(R.drawable.ic_time_dark);
+				}
+
 				break;
 
 			case PAST_COMMENTS:
-				menu.add(activity.getString(longText ? R.string.options_past_comments : R.string.options_past))
+				final MenuItem pastComments = menu.add(activity.getString(longText ? R.string.options_past_comments : R.string.options_past))
 						.setOnMenuItemClickListener(new MenuItem.OnMenuItemClickListener() {
 							public boolean onMenuItemClick(final MenuItem item) {
 								((OptionsMenuCommentsListener) activity).onPastComments();
 								return true;
 							}
 						});
+
+				if(showAsAction != MenuItem.SHOW_AS_ACTION_NEVER) {
+					pastComments.setShowAsAction(showAsAction);
+					pastComments.setIcon(R.drawable.ic_time_dark);
+				}
+
 				break;
 
 			case SUBSCRIBE:
-				menu.add(activity.getString(R.string.options_subscribe))
+				final MenuItem subscribe = menu.add(activity.getString(R.string.options_subscribe))
 						.setOnMenuItemClickListener(new MenuItem.OnMenuItemClickListener() {
 							public boolean onMenuItemClick(final MenuItem item) {
 								((OptionsMenuPostsListener)activity).onSubscribe();
 								return true;
 							}
 						});
+
+				subscribe.setShowAsAction(showAsAction);
+				subscribe.setIcon(R.drawable.star_off_dark);
+
 				break;
 
 			case UNSUBSCRIBE:
-				menu.add(activity.getString(R.string.options_unsubscribe))
+				final MenuItem unsubscribe = menu.add(activity.getString(R.string.options_unsubscribe))
```

Comment:
Just a minor note for next time: when declaring variables inside a switch statement, I usually put braces around each case, to stop the scope of the variables "leaking" out, and being accidentally used in a future case block. E.g:

```java
switch(var) {
    case 1: {
        int a = 5;
        break;
    }
    case 2: {
        // "a" is not accessible from here
    }
}
```

Reply:
Thanks. I put some braces in just to be safe.